File: 1002-find-common-characters/1002-find-common-characters.kt
class Solution {
fun commonChars(A: Array<String>): List<String> {
    val characters = Array(26) { IntArray(A.size) }
    for (i in A.indices)
        for (c in A[i])
            characters[c - 'a'][i]++
    return characters
        .withIndex()
        .map { (c, words) -> List(words.min()!!) { ('a' + c).toString() } }
        .flatten()
}
}

File: 1002-find-common-characters/1002-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        cnt = collections.Counter(words[0])
        for s in words:
            cnt2 = collections.Counter(s)
            for k in cnt.keys():
                cnt[k] = min(cnt[k], cnt2[k])
        return cnt.elements()
        